### ISSUE TITLE
Benchmark Serde impl for Proof & PreProcessedCircuit

### ISSUE THREAD

**Issue by CPerezz** [dusk-network/plonk#190]
Even serde implementation needed by #171 & #170 will be pretty fast since it's basically reading bytes, this benchmarks are interesting because:

- Verification time has to keep being low, and even serde will not affect it directly, it may affect the global Verification time within the DuskNetwork protocol. So it's important to check that the deserialization time for the `Proof` it's not too big compared to the magnitude of the `verify()` method.

- `Proof` serialization is also important (not as the previous statement) since the faster we serialize the `Proof`, the faster we're able to send it through the network to the block validators (Provisioners).

This benches could be added in #183

**Comment by CPerezz** [comment]
Benchmarks on Criterion for `PreProcessedCircuit` with a `circuit size` similar/equal to the Dusk one take an insane amount of time to be computed. Decreasing the sample size will not help. 

But the critical results were the `Proof` ones and we have them:
```
Proof Serde/Deserialization                                                                             
                        time:   [6.5842 ms 6.6738 ms 6.7583 ms]
Proof Serde/Serialization                                                                             
                        time:   [7.5748 us 7.7101 us 7.8419 us]
```
@toghrulmaharramov @kevaundray do you think it is acceptable? It takes almost the same tame to verify a `Proof` and de/serialize it. (Note that there's not too many things that we can optimize on this serde impl).

**Comment by CPerezz** [comment]
![image](https://user-images.githubusercontent.com/37264926/79556801-da5f7380-80a1-11ea-9bd1-711ac34e2a27.png)
Flamegraph part where the `Proof` deserialization is done. Not much to optimize if we don't go to `bls12_381` crate-level.

**Comment by CPerezz** [comment]
Closed atm via #183 

We already have an issue that focuses on improving these results. See #191
